### ISSUE TITLE
onBindViewHolder called un-necessarily?

### ISSUE THREAD

**Issue by rapid08** [TonicArtos/SuperSLiM#71]
I'm not sure if this is user error or if this expected behavior but I'm seeing onBindViewHolder being called more than it needs to be. I'm using a GridSLM and for every item at index changed, a large portion first column ( in a 3 column grid) gets updated. Some logs:

Example 1 -
Index changed: 31
Requesting index: 1
Requesting index: 2
Requesting index: 3
Requesting index: 21
Requesting index: 24
Requesting index: 27
Requesting index: 30
Requesting index: 31

I don't understand why the previous 4 row's first columns get updated as well as indexes 1-3.

Example 2 -
Index changed: 16
Requesting index: 4
Requesting index: 7
Requesting index: 13
Requesting index: 16
Requesting index: 16
Requesting index: 17
Requesting index: 17
Requesting index: 18
Requesting index: 19
Requesting index: 20
Requesting index: 21
Requesting index: 22
Requesting index: 23
Requesting index: 24
Requesting index: 25
Requesting index: 26
Requesting index: 27
Requesting index: 28
Requesting index: 29
Requesting index: 30
Requesting index: 31

I understand items 16-31 getting updated but why multiple times and why doe the indexes before 16 get updated?

**Comment by TonicArtos** [comment]
Some of this is definitely expected, some of it is probably my error. At least part of it is because SuperSLiM does not yet properly support data set changes. I very much doubt any of it is your fault. Finally, the development version is a work in progress to fix these issues..

FYI, for version 0.4, I am not worried much at all about this as I am trying to prepare for a new major version. However, I really do appreciate you bringing this up, it seems the situation is a little worse than I realised. As it turns out, supporting all the different possible configurations in SuperSLiM is extremely complex, so finding and reporting these overbinding issues is very helpful. Unfortunately, with the support for subsections that is introduced in the development version, this complexity has necessarily gone to a whole new level :(.

**Comment by bejibx** [comment]
Exact same behavior but with 2 column GridSLM. Looks like this:
```
[ 0      ] <-header
[ 1 ][ 2 ]
[ 3 ][ 4 ]
[ 5 ][ 6 ]
[ 7      ] <-header
...
```
Immediately after loading I'm getting following onBindViewHolders call log:
```
onBindViewHolder 0
onBindViewHolder 1
onBindViewHolder 2
onBindViewHolder 3
onBindViewHolder 3
onBindViewHolder 4
onBindViewHolder 5
onBindViewHolder 5
onBindViewHolder 6
onBindViewHolder 3
onBindViewHolder 5
onBindViewHolder 7
```

**Comment by TonicArtos** [comment]
@rapid08 Just letting you know that I haven't yet identified the cause for some of this behaviour, however I am working on tests to track it down. At the moment I think some of the onBind spam is caused by the need to look ahead for some kinds of nonsticky headers, some of this is planned for removal in 0.5.

**Comment by incandesco** [comment]
Hi Tonic, I think this may be caused by the same problem I have suggested a fix for at issue #114.  This was the behaviour I was seeing after a notifyItemChanged.  The suggested change seems to address this, at least for my usage.
